### PR TITLE
fix: yaml multiline string should be read successfully

### PR DESCRIPTION
**Pull request by noahnu**
The bug this fixes:

Adding

```
def test_multiline(snapshot):
    assert "multi\nline" == snapshot
```

to its own test file and running `inv test -u` would create a corrupted file.

### CODE REVIEW

**Comment by noahnu** [comment]
This is also an argument against custom parsers.

File: tests/test_snapshots.py [hunk 1]
```diff
@@ -12,6 +12,15 @@ def test_simple_string(snapshot):
     assert "Loreeeeeem ipsum." == snapshot
 
 
+def test_multiline(snapshot):
+    assert "multi\nline" == snapshot
+
+
+def test_singleline(snapshot):
+    assert "one-line" == snapshot
```
Comment (iamogbz) [inline tests/test_snapshots.py]: Not sure what this assertion adds, we already test single lines above. Is it the `-`?

Comment (iamogbz) [reply tests/test_snapshots.py]: Should also add a test case for python multi lines `"""..."""`

File: tests/test_snapshots.py [hunk 2]
```diff
@@ -12,6 +12,15 @@ def test_simple_string(snapshot):
...
+    assert "multi\nline 2\nline 3" == snapshot
```
Comment (iamogbz) [inline tests/test_snapshots.py]: Should also add a test case for python multi lines `"""..."""`

File: src/syrupy/serializers/base.py [hunk 3]
```diff
@@ -88,12 +88,13 @@ def create_or_update_snapshot(self, data: "SerializableData", index: int) -> Non
         self.write(data, index=index)
         self.post_write(data, index=index)
 
-    @final
+    @abstractmethod
     def delete_snapshot_from_file(self, snapshot_file: str, snapshot_name: str) -> None:
```
Comment (iamogbz) [inline src/syrupy/serializers/base.py]: This is now an explicit removal method

File: src/syrupy/serializers/raw_single.py [hunk 4]
```diff
@@ -44,13 +44,13 @@ def _read_file(self, filepath: str) -> Any:
...
 
-    def _write_snapshot_or_remove_file(
+    def _write_snapshot_to_file(
         self, snapshot_file: str, snapshot_name: str, data: "SerializableData"
     ) -> None:
-        if data:
-            self.__write_file(snapshot_file, data)
-        else:
-            os.remove(snapshot_file)
+        self.__write_file(snapshot_file, data)
```
Comment (iamogbz) [inline src/syrupy/serializers/raw_single.py]: Alway write snapshot data even if `None`

**Comment by noahnu** [comment]
Changes LGTM

**Pull request merged**